common-rest/src/main/java/org/uniprot/api/common/repository/search/SolrQueryConfigFileReader.java

```diff
@@ -0,0 +1,99 @@
+package org.uniprot.api.common.repository.search;
+
+import static org.uniprot.core.util.Utils.notNull;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.stream.Stream;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Created 04/09/19
+ *
+ * @author Edd
+ */
+@Slf4j
+public class SolrQueryConfigFileReader {
+    private static final String COMMENT_PREFIX = "#";
+    private final SolrQueryConfig.SolrQueryConfigBuilder builder;
+    private final String resourceLocation;
+
+    public SolrQueryConfigFileReader(String resourceLocation) {
+        this.resourceLocation = resourceLocation;
+        this.builder = SolrQueryConfig.builder();
+        initialiseSolrQueryConfig();
+    }
+
+    public SolrQueryConfig getConfig() {
+        return builder.build();
+    }
+
+    private void initialiseSolrQueryConfig() {
+        log.info("Loading Solr query config [" + this.resourceLocation + "]...");
+        InputStream resourceAsStream = getClass().getResourceAsStream(resourceLocation);
+        if (notNull(resourceAsStream)) {
+
+            Stream<String> lines =
+                    new BufferedReader(new InputStreamReader(resourceAsStream)).lines();
+            QueryConfigType queryConfigType = QueryConfigType.DEFAULT_SEARCH;
+            for (String line : lines.toArray(String[]::new)) {
```

Comment:
Edd, this is a suggestion/discussion, but, do you think we could use switch case here? 
because this prefix is the entire line, right? 
We would have to change from enum to constants do you think that it is worst?
What do you prefer?